### PR TITLE
feat: added docs for multicall [APE-1311]

### PR DESCRIPTION
**Pull request by Aviksaikat**
### What I did

fixes: #1611

### How I did it

Added documentation for `multicall` ape plugin.

### Checklist

- [ ] All changes are completed
- [ ] New test cases have been added
- [x] Documentation has been updated

### CODE REVIEW

File: docs/methoddocs/plugins.md [hunk 1]
```diff
@@ -69,3 +69,111 @@
...
+    :class:`~ape_ethereum.multicall.exceptions.UnsupportedChain`: If there is not an instance of Multicall3 deployed on the current chain at the expected address.
+
+Args:
+    **call_kwargs: the kwargs to pass through to the call handler.
+
+Returns:
+    Iterator[Any]: the sequence of values produced by performing each call stored by this instance.
+```
+
+#### Usage
```
Comment (antazoey) [inline docs/methoddocs/plugins.md:88]: In markdown, you need to only increase the number of `#` by one for each new hear.
The parent header has only two `#`s and you are going all the way up to four here.

File: docs/methoddocs/plugins.md [hunk 2]
```diff
@@ -69,3 +69,111 @@
     :members:
     :show-inheritance:
 ```
+
+## Multicall
+
+Ape comes with builtin `multicall` method. Yes you're reading this right. No need to install external modules to do multicall, ape got you covered.Perform the Multicall call. This call will trigger again every time the `Call` object is called.
```
Comment (antazoey) [inline docs/methoddocs/plugins.md:75]: Most of the words in the sentence are superfluous. I think only the first one (slightly re-written) is ok here.

```
The `ape-ethereum` core plugin comes with a multi-call module.
```

In general, docs should be simple and informative and not have any marketing terms.

**Review by antazoey — CHANGES_REQUESTED** [review]
this needs another round of revising.

File: docs/methoddocs/plugins.md [hunk 3]
```diff
@@ -69,3 +69,111 @@
...
+
+```js
```
[the remark this answers is not in the document]
Comment (Aviksaikat) [reply docs/methoddocs/plugins.md:77]: To make the syntax highlighting cool

File: docs/methoddocs/plugins.md [hunk 4]
```diff
@@ -69,3 +69,111 @@
     :members:
     :show-inheritance:
 ```
+
+## Multicall
```
[the remark this answers is not in the document]
Comment (Aviksaikat) [reply docs/methoddocs/plugins.md:73]: As `multicall` is under the `ape_ethereum` plugin I thought it should be under `plugins.md`. What do you think @fubuloubu ?

[the remark this answers is not in the document]
Comment (antazoey) [reply docs/methoddocs/plugins.md:73]: The plugins guide is talk about the plugin system, not each individual plugin.
However, calling out that is from the ape-ethereum core plugin I think is important.

[the remark this answers is not in the document]
Comment (Aviksaikat) [reply docs/methoddocs/plugins.md:73]: so should I put it in `transactions.md` ?

File: docs/methoddocs/plugins.md [hunk 5]
```diff
@@ -69,3 +69,111 @@
...
+
+Ape comes with builtin `multicall` method. Yes you're reading this right. No need to install external modules to do multicall, ape got you covered.Perform the Multicall call. This call will trigger again every time the `Call` object is called.
+
+```js
+Raises:
```
[the remark this answers is not in the document]
Comment (Aviksaikat) [reply docs/methoddocs/plugins.md:78]: Is it necessary? I can this repo https://github.com/banteg/multicall.py if you want

File: docs/methoddocs/plugins.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (antazoey) [reply docs/methoddocs/plugins.md:73]: Yes please.

[the remark this answers is not in the document]
Comment (antazoey) [reply docs/methoddocs/plugins.md:73]: Or `contracts.md` in the contract-interaction section.
But this plugins guide is meant only for describing the plugin system to users.

[the remark this answers is not in the document]
Comment (antazoey) [reply docs/methoddocs/plugins.md:73]: Actually--- this is not even the plugin's guide, this is the auto-generated method docs.
User guides should be in the user-guides.

File: docs/userguides/contracts.md [hunk 6]
```diff
@@ -224,3 +224,115 @@ contract = ape.Contract("0x...")
...
+    multicall_result = list(call())
+
+print(multicall_result[0])
+
+# output
+[17368643486106939361172, 31867695075486]
+```
+
+<!-- ### Encode Multicall Transaction
+Encode the Multicall transaction as a ``TransactionAPI`` object, but do not execute it.
```
Comment (antazoey) [inline docs/userguides/contracts.md:289]: not sure why this comment out, i dont think we need this section anyway

File: docs/userguides/contracts.md [hunk 7]
```diff
@@ -224,3 +224,115 @@ contract = ape.Contract("0x...")
...
+
+a, b, ..., z = call()  # Performs multicall
+# OR
+# The return type of the multicall is a generator object. So basically this will convert the result returned by the multicall into a list
+result = list(call()) 
+```
+
+### Practical Example
+
+This is a sample example of how you can perform multicall in a real world scenario. This piece of code will perfrom multicall on a `Uniswap V2` pool contract.
```
Comment (antazoey) [inline docs/userguides/contracts.md:269]: 1 sentence per line

File: docs/userguides/contracts.md [hunk 8]
```diff
@@ -224,3 +224,115 @@ contract = ape.Contract("0x...")
...
+...  # Add as many calls as desired
+call.add(contract.myMethod, *call_args)
+
+a, b, ..., z = call()  # Performs multicall
+# OR
+# The return type of the multicall is a generator object. So basically this will convert the result returned by the multicall into a list
+result = list(call()) 
+```
+
+### Practical Example
```
Comment (antazoey) [inline docs/userguides/contracts.md:267]: This section should probably be at the end

File: docs/userguides/contracts.md [hunk 9]
```diff
@@ -336,3 +292,25 @@ a, b, ..., z = txn(sender=my_signer)  # Sends the multical transaction
 # OR
 result = list(txn(sender=my_signer))
 ```
+
+### Practical Example
+
+This is a sample example of how you can perform multicall in a real world scenario.
```
Comment (antazoey) [inline docs/userguides/contracts.md:298]:
```suggestion
This is an example of how you can perform multicall in a real world scenario.
```

**Review by antazoey — DISMISSED** [review]

**Review by antazoey — APPROVED** [review]

File: docs/userguides/contracts.md [hunk 10]
```diff
@@ -224,3 +224,93 @@ contract = ape.Contract("0x...")
 bytes_value = contract.encode_input(0, 1, 2, 4, 5)
 method_id, input_dict = contract.decode_input(bytes_value)
 ```
+
+## Multicall
+
+The `ape_ethereum` core plugin comes with a `multi-call` module.
+
+```bash
+Raises:
```
Comment (antazoey) [inline docs/userguides/contracts.md:233]: method docs shouldnt be here

**Review by antazoey — DISMISSED** [review]
super close

**Comment by antazoey** [comment]
closing for https://github.com/ApeWorX/ape/pull/1649

**Pull request closed without merging**